cmd/object-handlers.go

```diff
@@ -2258,7 +2260,33 @@ func (api objectAPIHandlers) CompleteMultipartUploadHandler(w http.ResponseWrite
 	if api.CacheAPI() != nil {
 		completeMultiPartUpload = api.CacheAPI().CompleteMultipartUpload
 	}
+	completeDoneCh := make(chan struct{})
+	go func() {
+		// Send white spaces every 10 seconds to the client till completeMultiPartUpload() is done so that the client does not time out.
+		// Downside is we might send 200OK and then send error XML. But accoording to S3 spec
+		// the client is supposed to check for error XML even if it received 200OK. But for erasure this is not a
+		// problem as completeMultiPartUpload() is quick. Even For FS, it would not be an issue
+		// as we do background append as and when the parts arrive and completeMultiPartUpload is quick.
+		// Only in a rare case where parts would be out of order will FS' completeMultiPartUpload() take a longer time.
+		ticker := time.NewTicker(time.Second * 10)
+		for {
```

Comment:
We can perhaps do this in a function? @krishnasrinivas

Reply:
done